src/modules/uiTheme/components/provider/ThemeProvider.js

```diff
@@ -0,0 +1,56 @@
+import { BaElement } from '../../../BaElement';
+import { changeTheme } from '../../store/uiTheme.action';
+import { $injector } from '../../../../injection';
+
+/**
+ * Provides the body-element with a light or dark theme class
+ * @class
+ * @author aul
+ */
+export class ThemeProvider extends BaElement {
+
+	constructor() {
+		super();
+
+		const { EnvironmentService } = $injector.inject('EnvironmentService');
+		this._environmentService = EnvironmentService;
+	}
+
+	initialize() {
+
+		this._updateCss();
+
+		//listen to theme changes on window
+		this._environmentService.getWindow().matchMedia('(prefers-color-scheme: dark)').addEventListener('change', e => {
+			changeTheme(e.matches ? 'dark' : 'light');
+		});
+	}
+
+	_updateCss() {
+		
+		const { theme } = this._state;
+		const cssClassToAdd = theme === 'dark' ? 'dark-theme' : 'light-theme';
+		const cssClassToRemove = theme === 'light' ? 'dark-theme' : 'light-theme';
```

Comment:
Maybe a bit better readable: "const cssClassToAdd = (theme === 'dark') ? 'dark-theme' : 'light-theme';"

Reply:
We should let the linter ensure this. Configuration will follow in a separate PR...